### PR TITLE
APP-1577 Changed `approvals` and `minApprovals` and `addresslistLength` from `uint256` to `uint16`

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Description

Task: [APP-1577](https://aragonassociation.atlassian.net/browse/APP-1577)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-1577]: https://aragonassociation.atlassian.net/browse/APP-1577?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/contracts/contracts/voting/multisig/Multisig.sol [hunk 1]
```diff
@@ -173,10 +173,10 @@ contract Multisig is PluginUUPSUpgradeable, ProposalUpgradeable, Addresslist {
         _removeAddresses(_members);
 
         // Check if the new address list has become shorter than the current minimum number of approvals required.
-        uint256 newAddresslistLength = addresslistLength();
-        uint256 minApprovals_ = multisigSettings.minApprovals;
-        if (newAddresslistLength < minApprovals_) {
-            revert MinApprovalsOutOfBounds({limit: newAddresslistLength, actual: minApprovals_});
+        uint16 newAddresslistLength16 = addresslistLength().toUint16();
```
Comment (novaknole) [inline packages/contracts/contracts/voting/multisig/Multisig.sol:176]: I'd not transform it to uint16. If newAddresslistLength is 67,000 , and you convert it to uint16, its value will change which is incorrect.

File: packages/contracts/contracts/voting/multisig/Multisig.sol [hunk 2]
```diff
@@ -395,11 +395,11 @@ contract Multisig is PluginUUPSUpgradeable, ProposalUpgradeable, Addresslist {
 
     /// @notice Internal function to update the plugin settings.
     function _updateMultisigSettings(MultisigSettings calldata _multisigSettings) internal {
-        uint256 addresslistLength_ = addresslistLength();
+        uint16 addresslistLength16 = addresslistLength().toUint16();
```
Comment (novaknole) [inline packages/contracts/contracts/voting/multisig/Multisig.sol:398]: same as above.

File: packages/contracts/contracts/voting/multisig/Multisig.sol (hunk 1 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/multisig/Multisig.sol]: Fixed in 2e57532538654d7e3d0ed481b451c0f0cea759c3

Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/multisig/Multisig.sol]: Fixed in e434fe66a228351023cfaf9cbde8b8102695d63f

Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/multisig/Multisig.sol]: After a discussion with @novaknole (https://discord.com/channels/672466989217873929/1063135237699289150/1063503000632569936), we agreed to convert both, `minApprovals` and `addresslistLength`, to `uint32`.

File: packages/contracts/contracts/voting/multisig/Multisig.sol (hunk 2 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/multisig/Multisig.sol]: After a discussion with @novaknole (https://discord.com/channels/672466989217873929/1063135237699289150/1063503000632569936), we agreed to convert both, `minApprovals` and `addresslistLength`, to `uint32`.

**Comment by Michael-A-Heuer** [comment]
Duplicate of #195

**Pull request closed without merging**